### PR TITLE
Auto set CreatedAt for new DID documents

### LINKED ISSUE #3364: VDR v2 DID creation does not set CreatedAt for new documents
since all DID updates copy the value from the previous document version, the CreatedAt value is 0 for all versions

### PR DESCRIPTION
**Pull request by gerardsn**
closes #3364

### CODE REVIEW

File: vdr/didsubject/did_document_test.go [hunk 1]
```diff
@@ -76,18 +79,22 @@ func TestSqlDIDDocumentManager_CreateOrUpdate(t *testing.T) {
...
 		tx := db.Begin()
 		docManager := NewDIDDocumentManager(tx)
-		_, err := docManager.CreateOrUpdate(sqlDidBob, []orm.VerificationMethod{vm}, []orm.Service{service})
+		docRoot, err := docManager.CreateOrUpdate(sqlDidBob, []orm.VerificationMethod{vm}, []orm.Service{service})
 		require.NoError(t, err)
 		require.NoError(t, tx.Commit().Error)
 
 		docManager = NewDIDDocumentManager(transaction(t, db))
 		require.NoError(t, err)
+		time.Sleep(time.Second) // make sure doc.CreatedAt < doc.UpdatedAt
```
Comment (woutslakhorst) [inline vdr/didsubject/did_document_test.go:88]: try to prevent this. Could be done by setting a timeNow func in the package.

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: vdr/didsubject/did_document_test.go (hunk 1 above)
Comment (woutslakhorst) [reply vdr/didsubject/did_document_test.go:88]: but using gorm autoCreate makes this totally impossible...

Comment (woutslakhorst) [reply vdr/didsubject/did_document_test.go:88]: you could manually adjust the field value in the DB (low level) and then call update

**Comment by gerardsn** [comment]
Now also start counting document versions at 0 instead of 1

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**